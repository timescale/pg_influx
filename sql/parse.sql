CREATE EXTENSION influx;

-- Expect 2016-06-13T17:43:50.1004002Z (from protocol specification)
select * from parse_influx('cpu foo=12 1465839830100400200');

SELECT * FROM parse_influx(E'measurement,tag=foo field=12i 1465839830100400200');
SELECT * FROM parse_influx(E'measurement,tag=\\"foo field=12i 1465839830100400200');
SELECT * FROM parse_influx(E'measurement,tag=foo field=12i 1465839830100400200\nmeasurement,tag=bar field=12 1465839830100400200');
SELECT * FROM parse_influx(E'myMeasurement,tag1=value1,tag2=value2 fieldKey="fieldValue" 1556813561098000000');
SELECT * FROM parse_influx(E'myMeasurementName fieldKey="fieldValue" 1556813561098000000');
SELECT * FROM parse_influx(E'myMeasurementName fieldKey="fieldValue" 1556813561098000000\nmyMeasurement,tag1=value1,tag2=value2 fieldKey="fieldValue" 1556813561098000000');
SELECT * FROM parse_influx(E'disk,mode=rw,path=/boot/efi free=527806464i,inodes_free=i,total=0i,used_percent=1.4929822952022749 1574753954000000000');
SELECT * FROM parse_influx(E'disk,mode=0 free="527806464i",total=\\0i 1574753954000000000');
SELECT * FROM parse_influx(E'disk,mode=0,path=0i free="527806464i",total=\\0i 1574753954000000000');

\set ON_ERROR_STOP OFF
SELECT * FROM parse_influx(E'measurement,tag field=12 12345');
SELECT * FROM parse_influx(E'measurement, field=12 12345');
SELECT * FROM parse_influx(E'measurement 12345');
\set ON_ERROR_STOP ON

CREATE TABLE lines(line text);
COPY lines FROM stdin;
cpu,cpu=cpu0,host=fury usage_guest=0,usage_guest_nice=0,usage_idle=95.91836734330231,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=0,usage_steal=0,usage_system=2.0408163264927324,usage_user=2.0408163264927324 1574753954000000000
cpu,cpu=cpu1,host=fury usage_guest=0,usage_guest_nice=0,usage_idle=92.1568627436434,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=0,usage_steal=0,usage_system=3.921568627286635,usage_user=3.9215686274649673 1574753954000000000
cpu,cpu=cpu2,host=fury usage_guest=0,usage_guest_nice=0,usage_idle=90.74074073814575,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=1.8518518517645204,usage_steal=0,usage_system=3.703703703360616,usage_user=3.7037037035290408 1574753954000000000
cpu,cpu=cpu3,host=fury usage_guest=0,usage_guest_nice=0,usage_idle=96.00000000209548,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=0,usage_steal=0,usage_system=2.0000000000436557,usage_user=2.0000000000436557 1574753954000000000
cpu,cpu=cpu4,host=fury usage_guest=0,usage_guest_nice=0,usage_idle=91.83673469254418,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=0,usage_steal=0,usage_system=0,usage_user=8.163265305599708 1574753954000000000
cpu,cpu=cpu5,host=fury usage_guest=0,usage_guest_nice=0,usage_idle=97.95918367165116,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=0,usage_steal=0,usage_system=2.0408163264927324,usage_user=0 1574753954000000000
cpu,cpu=cpu6,host=fury usage_guest=0,usage_guest_nice=0,usage_idle=96.00000000186265,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=0,usage_steal=0,usage_system=0,usage_user=3.9999999999563443 1574753954000000000
cpu,cpu=cpu7,host=fury usage_guest=0,usage_guest_nice=0,usage_idle=94.00000000023283,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=0,usage_steal=0,usage_system=2.0000000000436557,usage_user=4.0000000000873115 1574753954000000000
cpu,cpu=cpu-total,host=fury usage_guest=0,usage_guest_nice=0,usage_idle=93.38235294084079,usage_iowait=0.24509803921656045,usage_irq=0,usage_nice=0,usage_softirq=0.4901960784331209,usage_steal=0,usage_system=1.9607843135541514,usage_user=3.921568627108303 1574753954000000000
disk,device=nvme0n1p2,fstype=ext4,host=fury,mode=rw,path=/ free=912578965504i,inodes_free=61635422i,inodes_total=62480384i,inodes_used=844962i,total=1006530654208i,used=42751348736i,used_percent=4.475033200428718 1574753954000000000
disk,device=nvme0n1p1,fstype=vfat,host=fury,mode=rw,path=/boot/efi free=527806464i,inodes_free=0i,inodes_total=0i,inodes_used=0i,total=535805952i,used=7999488i,used_percent=1.4929822952022749 1574753954000000000
diskio,host=fury,name=nvme0n1p2 io_time=2595292i,iops_in_progress=0i,read_bytes=309936706560i,read_time=8109392i,reads=6129866i,weighted_io_time=39313044i,write_bytes=80954531840i,write_time=37212815i,writes=2290766i 1574753954000000000
diskio,host=fury,name=loop1 io_time=108i,iops_in_progress=0i,read_bytes=1316864i,read_time=72i,reads=269i,weighted_io_time=32i,write_bytes=0i,write_time=0i,writes=0i 1574753954000000000
diskio,host=fury,name=loop4 io_time=20i,iops_in_progress=0i,read_bytes=351232i,read_time=5i,reads=47i,weighted_io_time=0i,write_bytes=0i,write_time=0i,writes=0i 1574753954000000000
diskio,host=fury,name=loop6 io_time=5184i,iops_in_progress=0i,read_bytes=50564096i,read_time=75687i,reads=48370i,weighted_io_time=68692i,write_bytes=0i,write_time=0i,writes=0i 1574753954000000000
diskio,host=fury,name=nvme0n1 io_time=2599684i,iops_in_progress=0i,read_bytes=309946164224i,read_time=8152676i,reads=6132370i,weighted_io_time=39387288i,write_bytes=80954532864i,write_time=37254690i,writes=2333085i 1574753954000000000
diskio,host=fury,name=nvme0n1p1 io_time=152i,iops_in_progress=0i,read_bytes=6955008i,read_time=42986i,reads=2397i,weighted_io_time=42616i,write_bytes=1024i,write_time=0i,writes=2i 1574753954000000000
diskio,host=fury,name=loop0 io_time=264i,iops_in_progress=0i,read_bytes=1100800i,read_time=51i,reads=779i,weighted_io_time=0i,write_bytes=0i,write_time=0i,writes=0i 1574753954000000000
diskio,host=fury,name=loop2 io_time=79952i,iops_in_progress=0i,read_bytes=493705216i,read_time=1409949i,reads=481841i,weighted_io_time=811912i,write_bytes=0i,write_time=0i,writes=0i 1574753954000000000
diskio,host=fury,name=loop3 io_time=216i,iops_in_progress=0i,read_bytes=1125376i,read_time=124i,reads=803i,weighted_io_time=76i,write_bytes=0i,write_time=0i,writes=0i 1574753954000000000
diskio,host=fury,name=loop5 io_time=204i,iops_in_progress=0i,read_bytes=2736128i,read_time=191i,reads=1672i,weighted_io_time=0i,write_bytes=0i,write_time=0i,writes=0i 1574753954000000000
kernel,host=fury boot_time=1574146313i,context_switches=1659702661i,entropy_avail=3797i,interrupts=838000993i,processes_forked=430825i 1574753954000000000
mem,host=fury active=8663875584i,available=2632663040i,available_percent=15.976348422300173,buffered=32174080i,cached=8126435328i,commit_limit=10386731008i,committed_as=26931036160i,dirty=864256i,free=2134585344i,high_free=0i,high_total=0i,huge_page_size=2097152i,huge_pages_free=0i,huge_pages_total=0i,inactive=4678496256i,low_free=0i,low_total=0i,mapped=5682855936i,page_tables=80105472i,shared=7086600192i,slab=595992576i,swap_cached=55185408i,swap_free=658169856i,swap_total=2147479552i,total=16478502912i,used=6185308160i,used_percent=37.53561954645604,vmalloc_chunk=0i,vmalloc_total=35184372087808i,vmalloc_used=0i,wired=0i,write_back=0i,write_back_tmp=0i 1574753954000000000
processes,host=fury blocked=0i,dead=0i,idle=68i,paging=0i,running=1i,sleeping=298i,stopped=0i,total=367i,total_threads=1363i,unknown=0i,zombies=0i 1574753954000000000
swap,host=fury free=658169856i,total=2147479552i,used=1489309696i,used_percent=69.3515193014513 1574753954000000000
swap,host=fury in=804974592i,out=6606921728i 1574753954000000000
system,host=fury load1=2.13,load15=0.84,load5=1.18,n_cpus=8i,n_users=1i 1574753954000000000
system,host=fury uptime=607641i 1574753954000000000
system,host=fury uptime_format="7 days,  0:47" 1574753954000000000
\.

SELECT * FROM (SELECT parse_influx(line) FROM lines) x;

DROP EXTENSION influx;
