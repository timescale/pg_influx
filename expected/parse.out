CREATE EXTENSION influx;
-- Expect 2016-06-13T17:43:50.1004002Z (from protocol specification)
select * from parse_influx('cpu foo=12 1465839830100400200');
 _metric |             _time             | _tags |    _fields    
---------+-------------------------------+-------+---------------
 cpu     | Mon Jun 13 17:43:50.1004 2016 | {}    | {"foo": "12"}
(1 row)

SELECT * FROM parse_influx(E'measurement,tag=foo field=12i 1465839830100400200');
   _metric   |             _time             |     _tags      |     _fields     
-------------+-------------------------------+----------------+-----------------
 measurement | Mon Jun 13 17:43:50.1004 2016 | {"tag": "foo"} | {"field": "12"}
(1 row)

SELECT * FROM parse_influx(E'measurement,tag=\\"foo field=12i 1465839830100400200');
   _metric   |             _time             |      _tags       |     _fields     
-------------+-------------------------------+------------------+-----------------
 measurement | Mon Jun 13 17:43:50.1004 2016 | {"tag": "\"foo"} | {"field": "12"}
(1 row)

SELECT * FROM parse_influx(E'measurement,tag=foo field=12i 1465839830100400200\nmeasurement,tag=bar field=12 1465839830100400200');
   _metric   |             _time             |     _tags      |     _fields     
-------------+-------------------------------+----------------+-----------------
 measurement | Mon Jun 13 17:43:50.1004 2016 | {"tag": "foo"} | {"field": "12"}
 measurement | Mon Jun 13 17:43:50.1004 2016 | {"tag": "bar"} | {"field": "12"}
(2 rows)

SELECT * FROM parse_influx(E'myMeasurement,tag1=value1,tag2=value2 fieldKey="fieldValue" 1556813561098000000');
    _metric    |            _time             |                _tags                 |          _fields           
---------------+------------------------------+--------------------------------------+----------------------------
 myMeasurement | Thu May 02 16:12:41.098 2019 | {"tag1": "value1", "tag2": "value2"} | {"fieldKey": "fieldValue"}
(1 row)

SELECT * FROM parse_influx(E'myMeasurementName fieldKey="fieldValue" 1556813561098000000');
      _metric      |            _time             | _tags |          _fields           
-------------------+------------------------------+-------+----------------------------
 myMeasurementName | Thu May 02 16:12:41.098 2019 | {}    | {"fieldKey": "fieldValue"}
(1 row)

SELECT * FROM parse_influx(E'myMeasurementName fieldKey="fieldValue" 1556813561098000000\nmyMeasurement,tag1=value1,tag2=value2 fieldKey="fieldValue" 1556813561098000000');
      _metric      |            _time             |                _tags                 |          _fields           
-------------------+------------------------------+--------------------------------------+----------------------------
 myMeasurementName | Thu May 02 16:12:41.098 2019 | {}                                   | {"fieldKey": "fieldValue"}
 myMeasurement     | Thu May 02 16:12:41.098 2019 | {"tag1": "value1", "tag2": "value2"} | {"fieldKey": "fieldValue"}
(2 rows)

SELECT * FROM parse_influx(E'disk,mode=rw,path=/boot/efi free=527806464i,inodes_free=i,total=0i,used_percent=1.4929822952022749 1574753954000000000');
 _metric |          _time           |                _tags                |                                            _fields                                            
---------+--------------------------+-------------------------------------+-----------------------------------------------------------------------------------------------
 disk    | Tue Nov 26 07:39:14 2019 | {"mode": "rw", "path": "/boot/efi"} | {"free": "527806464", "total": "0", "inodes_free": "i", "used_percent": "1.4929822952022749"}
(1 row)

SELECT * FROM parse_influx(E'disk,mode=0 free="527806464i",total=\\0i 1574753954000000000');
 _metric |          _time           |     _tags     |               _fields                
---------+--------------------------+---------------+--------------------------------------
 disk    | Tue Nov 26 07:39:14 2019 | {"mode": "0"} | {"free": "527806464i", "total": "0"}
(1 row)

SELECT * FROM parse_influx(E'disk,mode=0,path=0i free="527806464i",total=\\0i 1574753954000000000');
 _metric |          _time           |            _tags            |               _fields                
---------+--------------------------+-----------------------------+--------------------------------------
 disk    | Tue Nov 26 07:39:14 2019 | {"mode": "0", "path": "0i"} | {"free": "527806464i", "total": "0"}
(1 row)

\set ON_ERROR_STOP OFF
SELECT * FROM parse_influx(E'measurement,tag field=12 12345');
ERROR:  unexpected character
DETAIL:  expected '=' at position 15, saw ' '
SELECT * FROM parse_influx(E'measurement, field=12 12345');
ERROR:  identifier expected
DETAIL:  expected identifier at position 12, found ' '
SELECT * FROM parse_influx(E'measurement 12345');
ERROR:  identifier expected
DETAIL:  expected identifier at position 12, found '1'
\set ON_ERROR_STOP ON
CREATE TABLE lines(line text);
COPY lines FROM stdin;
SELECT * FROM (SELECT parse_influx(line) FROM lines) x;
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 parse_influx                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 (cpu,"Tue Nov 26 07:39:14 2019","{""cpu"": ""cpu0"", ""host"": ""fury""}","{""usage_irq"": ""0"", ""usage_idle"": ""95.91836734330231"", ""usage_nice"": ""0"", ""usage_user"": ""2.0408163264927324"", ""usage_guest"": ""0"", ""usage_steal"": ""0"", ""usage_iowait"": ""0"", ""usage_system"": ""2.0408163264927324"", ""usage_softirq"": ""0"", ""usage_guest_nice"": ""0""}")
 (cpu,"Tue Nov 26 07:39:14 2019","{""cpu"": ""cpu1"", ""host"": ""fury""}","{""usage_irq"": ""0"", ""usage_idle"": ""92.1568627436434"", ""usage_nice"": ""0"", ""usage_user"": ""3.9215686274649673"", ""usage_guest"": ""0"", ""usage_steal"": ""0"", ""usage_iowait"": ""0"", ""usage_system"": ""3.921568627286635"", ""usage_softirq"": ""0"", ""usage_guest_nice"": ""0""}")
 (cpu,"Tue Nov 26 07:39:14 2019","{""cpu"": ""cpu2"", ""host"": ""fury""}","{""usage_irq"": ""0"", ""usage_idle"": ""90.74074073814575"", ""usage_nice"": ""0"", ""usage_user"": ""3.7037037035290408"", ""usage_guest"": ""0"", ""usage_steal"": ""0"", ""usage_iowait"": ""0"", ""usage_system"": ""3.703703703360616"", ""usage_softirq"": ""1.8518518517645204"", ""usage_guest_nice"": ""0""}")
 (cpu,"Tue Nov 26 07:39:14 2019","{""cpu"": ""cpu3"", ""host"": ""fury""}","{""usage_irq"": ""0"", ""usage_idle"": ""96.00000000209548"", ""usage_nice"": ""0"", ""usage_user"": ""2.0000000000436557"", ""usage_guest"": ""0"", ""usage_steal"": ""0"", ""usage_iowait"": ""0"", ""usage_system"": ""2.0000000000436557"", ""usage_softirq"": ""0"", ""usage_guest_nice"": ""0""}")
 (cpu,"Tue Nov 26 07:39:14 2019","{""cpu"": ""cpu4"", ""host"": ""fury""}","{""usage_irq"": ""0"", ""usage_idle"": ""91.83673469254418"", ""usage_nice"": ""0"", ""usage_user"": ""8.163265305599708"", ""usage_guest"": ""0"", ""usage_steal"": ""0"", ""usage_iowait"": ""0"", ""usage_system"": ""0"", ""usage_softirq"": ""0"", ""usage_guest_nice"": ""0""}")
 (cpu,"Tue Nov 26 07:39:14 2019","{""cpu"": ""cpu5"", ""host"": ""fury""}","{""usage_irq"": ""0"", ""usage_idle"": ""97.95918367165116"", ""usage_nice"": ""0"", ""usage_user"": ""0"", ""usage_guest"": ""0"", ""usage_steal"": ""0"", ""usage_iowait"": ""0"", ""usage_system"": ""2.0408163264927324"", ""usage_softirq"": ""0"", ""usage_guest_nice"": ""0""}")
 (cpu,"Tue Nov 26 07:39:14 2019","{""cpu"": ""cpu6"", ""host"": ""fury""}","{""usage_irq"": ""0"", ""usage_idle"": ""96.00000000186265"", ""usage_nice"": ""0"", ""usage_user"": ""3.9999999999563443"", ""usage_guest"": ""0"", ""usage_steal"": ""0"", ""usage_iowait"": ""0"", ""usage_system"": ""0"", ""usage_softirq"": ""0"", ""usage_guest_nice"": ""0""}")
 (cpu,"Tue Nov 26 07:39:14 2019","{""cpu"": ""cpu7"", ""host"": ""fury""}","{""usage_irq"": ""0"", ""usage_idle"": ""94.00000000023283"", ""usage_nice"": ""0"", ""usage_user"": ""4.0000000000873115"", ""usage_guest"": ""0"", ""usage_steal"": ""0"", ""usage_iowait"": ""0"", ""usage_system"": ""2.0000000000436557"", ""usage_softirq"": ""0"", ""usage_guest_nice"": ""0""}")
 (cpu,"Tue Nov 26 07:39:14 2019","{""cpu"": ""cpu-total"", ""host"": ""fury""}","{""usage_irq"": ""0"", ""usage_idle"": ""93.38235294084079"", ""usage_nice"": ""0"", ""usage_user"": ""3.921568627108303"", ""usage_guest"": ""0"", ""usage_steal"": ""0"", ""usage_iowait"": ""0.24509803921656045"", ""usage_system"": ""1.9607843135541514"", ""usage_softirq"": ""0.4901960784331209"", ""usage_guest_nice"": ""0""}")
 (disk,"Tue Nov 26 07:39:14 2019","{""host"": ""fury"", ""mode"": ""rw"", ""path"": ""/"", ""device"": ""nvme0n1p2"", ""fstype"": ""ext4""}","{""free"": ""912578965504"", ""used"": ""42751348736"", ""total"": ""1006530654208"", ""inodes_free"": ""61635422"", ""inodes_used"": ""844962"", ""inodes_total"": ""62480384"", ""used_percent"": ""4.475033200428718""}")
 (disk,"Tue Nov 26 07:39:14 2019","{""host"": ""fury"", ""mode"": ""rw"", ""path"": ""/boot/efi"", ""device"": ""nvme0n1p1"", ""fstype"": ""vfat""}","{""free"": ""527806464"", ""used"": ""7999488"", ""total"": ""535805952"", ""inodes_free"": ""0"", ""inodes_used"": ""0"", ""inodes_total"": ""0"", ""used_percent"": ""1.4929822952022749""}")
 (diskio,"Tue Nov 26 07:39:14 2019","{""host"": ""fury"", ""name"": ""nvme0n1p2""}","{""reads"": ""6129866"", ""writes"": ""2290766"", ""io_time"": ""2595292"", ""read_time"": ""8109392"", ""read_bytes"": ""309936706560"", ""write_time"": ""37212815"", ""write_bytes"": ""80954531840"", ""iops_in_progress"": ""0"", ""weighted_io_time"": ""39313044""}")
 (diskio,"Tue Nov 26 07:39:14 2019","{""host"": ""fury"", ""name"": ""loop1""}","{""reads"": ""269"", ""writes"": ""0"", ""io_time"": ""108"", ""read_time"": ""72"", ""read_bytes"": ""1316864"", ""write_time"": ""0"", ""write_bytes"": ""0"", ""iops_in_progress"": ""0"", ""weighted_io_time"": ""32""}")
 (diskio,"Tue Nov 26 07:39:14 2019","{""host"": ""fury"", ""name"": ""loop4""}","{""reads"": ""47"", ""writes"": ""0"", ""io_time"": ""20"", ""read_time"": ""5"", ""read_bytes"": ""351232"", ""write_time"": ""0"", ""write_bytes"": ""0"", ""iops_in_progress"": ""0"", ""weighted_io_time"": ""0""}")
 (diskio,"Tue Nov 26 07:39:14 2019","{""host"": ""fury"", ""name"": ""loop6""}","{""reads"": ""48370"", ""writes"": ""0"", ""io_time"": ""5184"", ""read_time"": ""75687"", ""read_bytes"": ""50564096"", ""write_time"": ""0"", ""write_bytes"": ""0"", ""iops_in_progress"": ""0"", ""weighted_io_time"": ""68692""}")
 (diskio,"Tue Nov 26 07:39:14 2019","{""host"": ""fury"", ""name"": ""nvme0n1""}","{""reads"": ""6132370"", ""writes"": ""2333085"", ""io_time"": ""2599684"", ""read_time"": ""8152676"", ""read_bytes"": ""309946164224"", ""write_time"": ""37254690"", ""write_bytes"": ""80954532864"", ""iops_in_progress"": ""0"", ""weighted_io_time"": ""39387288""}")
 (diskio,"Tue Nov 26 07:39:14 2019","{""host"": ""fury"", ""name"": ""nvme0n1p1""}","{""reads"": ""2397"", ""writes"": ""2"", ""io_time"": ""152"", ""read_time"": ""42986"", ""read_bytes"": ""6955008"", ""write_time"": ""0"", ""write_bytes"": ""1024"", ""iops_in_progress"": ""0"", ""weighted_io_time"": ""42616""}")
 (diskio,"Tue Nov 26 07:39:14 2019","{""host"": ""fury"", ""name"": ""loop0""}","{""reads"": ""779"", ""writes"": ""0"", ""io_time"": ""264"", ""read_time"": ""51"", ""read_bytes"": ""1100800"", ""write_time"": ""0"", ""write_bytes"": ""0"", ""iops_in_progress"": ""0"", ""weighted_io_time"": ""0""}")
 (diskio,"Tue Nov 26 07:39:14 2019","{""host"": ""fury"", ""name"": ""loop2""}","{""reads"": ""481841"", ""writes"": ""0"", ""io_time"": ""79952"", ""read_time"": ""1409949"", ""read_bytes"": ""493705216"", ""write_time"": ""0"", ""write_bytes"": ""0"", ""iops_in_progress"": ""0"", ""weighted_io_time"": ""811912""}")
 (diskio,"Tue Nov 26 07:39:14 2019","{""host"": ""fury"", ""name"": ""loop3""}","{""reads"": ""803"", ""writes"": ""0"", ""io_time"": ""216"", ""read_time"": ""124"", ""read_bytes"": ""1125376"", ""write_time"": ""0"", ""write_bytes"": ""0"", ""iops_in_progress"": ""0"", ""weighted_io_time"": ""76""}")
 (diskio,"Tue Nov 26 07:39:14 2019","{""host"": ""fury"", ""name"": ""loop5""}","{""reads"": ""1672"", ""writes"": ""0"", ""io_time"": ""204"", ""read_time"": ""191"", ""read_bytes"": ""2736128"", ""write_time"": ""0"", ""write_bytes"": ""0"", ""iops_in_progress"": ""0"", ""weighted_io_time"": ""0""}")
 (kernel,"Tue Nov 26 07:39:14 2019","{""host"": ""fury""}","{""boot_time"": ""1574146313"", ""interrupts"": ""838000993"", ""entropy_avail"": ""3797"", ""context_switches"": ""1659702661"", ""processes_forked"": ""430825""}")
 (mem,"Tue Nov 26 07:39:14 2019","{""host"": ""fury""}","{""free"": ""2134585344"", ""slab"": ""595992576"", ""used"": ""6185308160"", ""dirty"": ""864256"", ""total"": ""16478502912"", ""wired"": ""0"", ""active"": ""8663875584"", ""cached"": ""8126435328"", ""mapped"": ""5682855936"", ""shared"": ""7086600192"", ""buffered"": ""32174080"", ""inactive"": ""4678496256"", ""low_free"": ""0"", ""available"": ""2632663040"", ""high_free"": ""0"", ""low_total"": ""0"", ""swap_free"": ""658169856"", ""high_total"": ""0"", ""swap_total"": ""2147479552"", ""write_back"": ""0"", ""page_tables"": ""80105472"", ""swap_cached"": ""55185408"", ""commit_limit"": ""10386731008"", ""committed_as"": ""26931036160"", ""used_percent"": ""37.53561954645604"", ""vmalloc_used"": ""0"", ""vmalloc_chunk"": ""0"", ""vmalloc_total"": ""35184372087808"", ""huge_page_size"": ""2097152"", ""write_back_tmp"": ""0"", ""huge_pages_free"": ""0"", ""huge_pages_total"": ""0"", ""available_percent"": ""15.976348422300173""}")
 (processes,"Tue Nov 26 07:39:14 2019","{""host"": ""fury""}","{""dead"": ""0"", ""idle"": ""68"", ""total"": ""367"", ""paging"": ""0"", ""blocked"": ""0"", ""running"": ""1"", ""stopped"": ""0"", ""unknown"": ""0"", ""zombies"": ""0"", ""sleeping"": ""298"", ""total_threads"": ""1363""}")
 (swap,"Tue Nov 26 07:39:14 2019","{""host"": ""fury""}","{""free"": ""658169856"", ""used"": ""1489309696"", ""total"": ""2147479552"", ""used_percent"": ""69.3515193014513""}")
 (swap,"Tue Nov 26 07:39:14 2019","{""host"": ""fury""}","{""in"": ""804974592"", ""out"": ""6606921728""}")
 (system,"Tue Nov 26 07:39:14 2019","{""host"": ""fury""}","{""load1"": ""2.13"", ""load5"": ""1.18"", ""load15"": ""0.84"", ""n_cpus"": ""8"", ""n_users"": ""1""}")
 (system,"Tue Nov 26 07:39:14 2019","{""host"": ""fury""}","{""uptime"": ""607641""}")
 (system,"Tue Nov 26 07:39:14 2019","{""host"": ""fury""}","{""uptime_format"": ""7 days,  0:47""}")
(29 rows)

DROP EXTENSION influx;
